CONTRIBUTING.rst

```diff
@@ -186,15 +186,48 @@ WMCore defines its own pylintrc `here <https://github.com/dmwm/WMCore/blob/maste
 Thus, when evaluating your changes, please run pylint by passing this pylintrc file in the command line, your code should get scored 8 or above.
 Unless there is a very good reason, we discourage the use of pylint disable statements.
 
+Project Docstrings Best Practices
+---------------------------------
+
+With the goal of uniformizing and making the project more readable, we are adopting the `Sphinx` docstring style, which uses
+reStructuredText markup. It's meant to document entire modules, classes, methods and functions. To avoid too large docstrings, we propose a subtle variation of this style such that each parameter
```

Comment:
If you're serious about enforcing rst format, then you should at least provide references to its documentation. In particular the references should include the following items (but not limited to them):
- format description, like usage of `===` instead of `### for headers, etc.
- special characters, like usage of `*` or `_`
- space separations for parameters and main text
- how to make lists, etc.

As we learnt (you struggle with rst format recently if I recall correctly), it is very easy to mix different formats from Markdown and rst? Said that, apart from rst documentation, I suggest adding additional local document to demonstrate all use-case (I listed above and others) which developers may encounter. Searching for proper data-format takes time, and single document which will list how to write in rst would be very helpful. Such document should provide examples with functions and their docstring, etc.

Reply:
That's a fair comment. I will provide further links in my next commit.

However, I do think we should keep docstrings as simple as possible. If anything requires too much embellishment and heavy use of the markup syntax, then we are probably doing it wrong! We should set a wiki in the documentation and have a reference for it in the docstring.